### PR TITLE
feat: improve CI workflow

### PR DESCRIPTION
**Pull request by RafilxTenfen**
## Description

- Added cache for umeed v103 and mainnet tinkered genesis
- Add dependencies between workflows for already built binaries for linux amd64
- removed deprecated rokroskar/workflow-run-cleanup-action@master 

__Overview of tests.yml__
image

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by RafilxTenfen** [comment]
- Added steps to cache tinkered genesis and umeed-v103. Cache stored at https://github.com/umee-network/umee/actions/runs/3040108775/jobs/4895779303
- Cache used in the next run https://github.com/umee-network/umee/actions/runs/3040108775/jobs/4895978516

**Comment by RafilxTenfen** [comment]
@robert-zaremba @toteki anyone against merging it?

File: contrib/scripts/tikerer-mainnet-genesis.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash -eu
+
+# tikerer the mainnet genesis by swapping the polychain by priv_validator_key
+
+# USAGE: ./tikerer-mainnet-genesis.sh
```
Comment (toteki) [inline contrib/scripts/tikerer-mainnet-genesis.sh:5]:
```suggestion
# tinker the mainnet genesis by swapping polychain -> priv_validator_key

# USAGE: ./tinker-mainnet-genesis.sh
```
Plus: rename file to `tinker-mainnet-genesis.sh`

**Comment by toteki** [comment]
Looks good - I didn't test any local runs though. Commented on some typos

**Comment by RafilxTenfen** [comment]
@toteki tested CI again at https://github.com/umee-network/umee/actions/runs/3054312810

**Review by khoerling — APPROVED** [review]
Pre-approval

**Pull request merged**